src/core/Akka.Remote/Transport/DotNetty/DotNettyTransportSettings.cs

```diff
@@ -0,0 +1,185 @@
+//  <copyright file="DotNettyTransportSettings.cs" company="Akka.NET project">
+//      Copyright (C) 2009-2016 Lightbend Inc. <http://www.lightbend.com>
+//      Copyright (C) 2013-2017 Akka.NET project <https://github.com/akkadotnet>
+//  </copyright>
+// -----------------------------------------------------------------------
+#endregion
+
+using System;
+using System.Linq;
+using System.Net;
+using System.Security.Cryptography.X509Certificates;
+using Akka.Actor;
+using Akka.Configuration;
+using Akka.Dispatch;
+using Akka.Util;
+using DotNetty.Buffers;
+
+namespace Akka.Remote.Transport.DotNetty
+{
+    internal sealed class DotNettyTransportSettings
+    {
+        public static DotNettyTransportSettings Create(Config config)
+        {
+            var transportMode = config.GetString("transport-protocol", "tcp").ToLower();
+            var host = config.GetString("hostname", null);
+            if (string.IsNullOrEmpty(host)) host = IPAddress.Any.ToString();
+            var publicHost = config.GetString("public-hostname", null);
+
+            var order = ByteOrder.LittleEndian;
+            var byteOrderString = config.GetString("byte-order", "little-endian").ToLowerInvariant();
+            switch (byteOrderString)
+            {
+                case "little-endian": order = ByteOrder.LittleEndian; break;
+                case "big-endian": order = ByteOrder.BigEndian; break;
+                default: throw new ArgumentException($"Unknown byte-order option [{byteOrderString}]. Supported options are: big-endian, little-endian.");
+            }
+
+            return new DotNettyTransportSettings(
+                transportMode: transportMode == "tcp" ? TransportMode.Tcp : TransportMode.Udp,
+                enableSsl: config.GetBoolean("enable-ssl", false),
+                connectTimeout: config.GetTimeSpan("connection-timeout", TimeSpan.FromSeconds(15)),
+                hostname: host,
+                publicHostname: !string.IsNullOrEmpty(publicHost) ? publicHost : host,
+                port: config.GetInt("port", 2552),
+                serverSocketWorkerPoolSize: ComputeWorkerPoolSize(config.GetConfig("server-socket-worker-pool")),
+                clientSocketWorkerPoolSize: ComputeWorkerPoolSize(config.GetConfig("client-socket-worker-pool")),
+                maxFrameSize: ToNullableInt(config.GetByteSize("maximum-frame-size")) ?? 128000,
+                ssl: config.HasPath("ssl") ? SslSettings.Create(config.GetConfig("ssl")) : SslSettings.Empty,
+                dnsUseIpv6: config.GetBoolean("dns-use-ipv6", false),
+                tcpReuseAddr: config.GetBoolean("tcp-reuse-addr", true),
+                tcpKeepAlive: config.GetBoolean("tcp-keepalive", true),
+                tcpNoDelay: config.GetBoolean("tcp-nodelay", true),
+                backlog: config.GetInt("backlog", 4096),
+                enforceIpFamily: RuntimeDetector.IsMono || config.GetBoolean("enforce-ip-family", false),
```

Comment:
I haven't checked yet, but do we know if the newer (4.6+) versions of Mono function correctly with IPV6 now?